src/components/fetchhelper.ts

```diff
@@ -20,49 +20,49 @@ export function getFetchPromise(request) {
     if (contentType) {
         headers['Content-Type'] = contentType;
     }
-    var url = request.url;
+    let url = request.url;
     if (request.query) {
-        var paramString = paramsToString(request.query);
+        const paramString = paramsToString(request.query);
         paramString && (url += '?' + paramString);
     }
     return request.timeout
         ? fetchWithTimeout(url, fetchRequest, request.timeout)
         : fetch(url, fetchRequest);
 }
 
-export function fetchWithTimeout(url, options, timeoutMs) {
-    return (
-        console.log(
-            'fetchWithTimeout: timeoutMs: ' + timeoutMs + ', url: ' + url
-        ),
-        new Promise(function (resolve, reject) {
-            var timeout = setTimeout(reject, timeoutMs);
-            options = options || {};
-            options.credentials = 'same-origin';
-            fetch(url, options).then(
-                function (response) {
-                    clearTimeout(timeout);
-                    console.log(
-                        'fetchWithTimeout: succeeded connecting to url: ' + url
-                    );
-                    resolve(response);
-                },
-                function () {
-                    clearTimeout(timeout);
-                    console.log(
-                        'fetchWithTimeout: timed out connecting to url: ' + url
-                    );
-                    reject();
-                }
-            );
-        })
-    );
+export function fetchWithTimeout(
+    url: string,
+    options: any,
+    timeoutMs: number
+): Promise<any> {
```

Comment:
```suggestion
export function fetchWithTimeout(
    url: string,
    options: Record<string, string>,
    timeoutMs: number
): Promise<void> {
```

Same as above. I'm on phone and can't say from memory, but iirc, this should be a dictionary with keys and values of type string, so my suggestion applies. If not, send some examples of possible dicts and I can help you get the correct type.

Also, use ``void`` as return value on functions that doesn't return anything.

Reply:
It returns stuff.
I have started modelling the request type because it's a structure with a limited scope. I'm confident enough to refactor that now

Reply:
And I think options.headers is its own dictionary.

Reply:
I did find better types for this function